src/components/Dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+
+interface Props {
+  people: Person[];
+  delay: number;
+  onSelected?: (user: Person) => void,
+}
+
+export const Dropdown: React.FC<Props> = React.memo(({
+  people,
+  onSelected = () => {},
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [hasFocus, setHasFocus] = useState(false);
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+    applyQuery(event.target.value);
+    if (query) {
+      setIsProcessing(true);
+    }
```

Comment:
You got the idea of debounce wrong

You should not emulate the call to the server - rather it limits the number of calls to the underlying API

So you don't need  `isProcessing` state at all

Most autocomplete components just show the current list, while the person is typing and update the value when the user stops typing (i.e. debounce the call)